### PR TITLE
Allow NetPlay's Initial GCTime to be set

### PR DESCRIPTION
**Pull request by SpencerAlanWatson**
Fixed [Issue #8030](https://code.google.com/p/dolphin-emu/issues/detail?id=8030)

So, these changes allows people to set the initial GCTime for NetPlay, which is the time netplay sets for everyone to keep them sync'd.  How it works, is that the server will load the time from the ini, and then send to the clients when they connect. 

I have not added this to the GUI in any way, as I do not know if it should be in the GUI.  I feel it might just confuse people.

### CODE REVIEW

File: Source/Core/Core/ConfigManager.h [hunk 1]
```diff
@@ -139,6 +142,8 @@ struct SConfig : NonCopyable
 	void SaveInputSettings(IniFile& ini);
 	void SaveMovieSettings(IniFile& ini);
 	void SaveFifoPlayerSettings(IniFile& ini);
+	void SaveNetPlaySettings(IniFile& ini);
+
```
Comment (Buddybenj) [inline Source/Core/Core/ConfigManager.h]: Extraneous new line.

File: Source/Core/Core/NetPlayClient.cpp [hunk 2]
```diff
@@ -27,7 +27,10 @@ NetPlayClient::~NetPlayClient()
 	// not perfect
 	if (m_is_running)
 		StopGame();
-
+	
+	// incase it was changed by the server
```
Comment (Buddybenj) [inline Source/Core/Core/NetPlayClient.cpp]: Can you make this properly capitalized?

File: Source/Core/Core/NetPlayClient.cpp [hunk 3]
```diff
@@ -842,7 +853,7 @@ u32 CEXIIPL::NetPlay_GetGCTime()
 	std::lock_guard<std::mutex> lk(crit_netplay_client);
 
 	if (netplay_client)
-		return NETPLAY_INITIAL_GCTIME; // watev
+		return g_NetplayInitialGCTime; // watev
```
Comment (Buddybenj) [inline Source/Core/Core/NetPlayClient.cpp]: I don't think the "watev" comment is needed anymore. :P

**Comment by Buddybenj** [comment]
I don't understand what you're doing with the first three commits. They aren't supposed to be there right?

File: Source/Core/Core/NetPlayProto.h [hunk 4]
```diff
@@ -30,7 +30,7 @@ typedef std::vector<u8> NetWiimote;
 
 #define NETPLAY_VERSION  "Dolphin NetPlay 2014-01-08"
 
-const int NETPLAY_INITIAL_GCTIME = 1272737767;
+extern int g_NetplayInitialGCTime = 1272737767;
```
Comment (Armada651) [inline Source/Core/Core/NetPlayProto.h]: Why mark it as `extern` when you don't declare it anywhere else?

**Comment by JMC47** [comment]
Please fix the comments and then I'll have buildbot build it and test it out myself.

**Comment by Stevoisiak** [comment]
@dolphin-emu-bot rebuild

**Comment by skidau** [comment]
Superceded by #2183

**Pull request closed without merging**